### PR TITLE
Add leetcode.com contest list

### PR DESCRIPTION
**Pull request by a180285**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/leetcode-com/contest
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [x] New Route
- [x] Documentation
  - [x] CN
  - [ ] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] 日期和时间 date and time
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [x] 添加了新的包 New package added 
  - moment.js 用于格式化时间字符串
- [ ] `Puppeteer`

## 说明 / Note
使用了 leetcode-com 与方便后期与 leetcode.cn 进行区分

### CODE REVIEW

File: lib/v2/leetcode-com/contest.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    });
+    const allContests = response.data.data.allContests;
+
+    const contestBaseUrl = 'https://leetcode.com/contest/';
+    const items = allContests.slice(0, 10).map((contest) => {
+        const beijingTime = moment
+            .utc(contest.startTime * 1000)
+            .utcOffset('+0800')
+            .format('YYYY-MM-DD HH:mm:ss');
+        const duration = contest.duration / 3600.0;
```
Comment (TonyRL) [inline lib/v2/leetcode-com/contest.js:22]: I think you can also use day.js to achieve which is already available in current package.json

File: lib/v2/leetcode-com/contest.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        return {
+            title: contest.title + ` ( ${beijingTime}, ${duration} 小时 )`,
+            description: `北京时间： ${beijingTime} ，时长： ${duration} 小时<br>` + contest.description,
```
Comment (TonyRL) [inline lib/v2/leetcode-com/contest.js:25]: Please use art template engine.
See https://github.com/DIYgod/RSSHub/discussions/8359
Docs: https://docs.rsshub.app/joinus/script-standard.html#template

File: docs/programming.md [hunk 3]
```diff
@@ -292,6 +292,10 @@ GitHub 官方也提供了一些 RSS:
 
 <Route author="NathanDai" example="/leetcode/submission/us/nathandai" path="/leetcode/submission/:country/:user" :paramsDesc="['国家 country, 中国(cn)和美国(us)', '用户名 user, 可在LeetCode用户主页的 URL 中找到']"/>
 
+## leetcode.com
```
[the remark this answers is not in the document]
Comment (TonyRL) [reply docs/programming.md:295]: There's already a H2 header of LeetCode few lines above.

Please also add english docs.

**Review by TonyRL — CHANGES_REQUESTED** [review]
I would suggest using the namespace of `leetcode` instead of `leetcode-com`.

**Comment by a180285** [comment]
Waiting for #8787

**Pull request closed without merging**